modules/ROOT/pages/constraints/managing-constraints.adoc

```diff
@@ -0,0 +1,1598 @@
+
+This applies to all constraint types.
+
+.Create a node that violates an existing node property uniqueness constraint
+======
+
+.Create a `Book` node with an `isbn` that is already used in the graph
+[source, cypher, role=test-fail]
+----
+CREATE (book:Book {isbn: '1449356265', title: 'Graph Databases'})
+----
+
+In this case, the node is not created because the `isbn` property is in conflict with an existing constraint.
+
+.Error message
+[source, error]
+----
+Node(0) already exists with label `Book` and property `isbn` = '1449356265'
+----
+
+======
+
+.Create a node that violates an existing node property existence constraint
+======
+
+.Create an `Author` node without a `name` property, given a property existence constraint on `:Author(name)`
+[source, cypher, role=test-fail]
+----
+CREATE (author:Author {surname: 'Austen'})
+----
+
+In this case, the node is not created because it is missing the `name` property which is in conflict with an existing constraint.
+
+.Error message
+[source, error]
+----
+Node(0) with label `Author` must have the property `name`
+----
+
+======
+
+.Create a relationship that violates an existing relationship property type constraint
+======
+
+
+.Create a `PART_OF` relationship with a `STRING` `order` property, given a property type constraint on the relationship type `PART_OF` restricting the `order` property to `INTEGER` values
+[source, cypher, role=test-fail]
+----
+MATCH (movie:Movie {title:'Iron Man'}), (franchise:Franchise {name:'MCU'})
+CREATE (movie)-[part:PART_OF {order: '1'}]->(franchise)
+----
+
+In this case, the relationship is not created because the `order` property is in conflict with an existing constraint.
+
+.Error message
+[source, error]
+----
+Relationship(0) with type `PART_OF` has property `order` of wrong type `String`. Allowed types: INTEGER
+----
+
+======
+
+
+.Create a node that violates existing node property key constraint
+======
+
+.Create an `Actor` node without a `firstname` property, given a node key constraint on `:Actor(firstname, surname)`
+[source, cypher, role=test-fail]
+----
+CREATE (actor:Actor {surname: 'Wood'})
+----
+
+In this case, the node is not created because it is missing the `firstname` property which is in conflict with an existing constraint.
+
+.Error message
+[source, error]
+----
+Node(0) with label `Actor` must have the properties (`firstname`, `surname`)
+----
+
+======
+
+[[fail-to-create-constraint-due-to-existing-data]]
+==== Creating constraints when there exists conflicting data will fail
+
+This applies to all constraint types.
+
+.Create a node property uniqueness constraint when conflicting nodes exist
+======
+
+
+////
+[source, cypher, role=test-setup]
+----
+CREATE (book:Book {isbn: '9780393972832', title: 'Moby Dick'});
+CREATE (book:Book {isbn: '9780763630188', title: 'Moby Dick'})
+----
+////
+
+.Create a constraint requiring `Book` nodes to have unique `title` properties, when there already exists two nodes with the same `title`
```

Comment:
do we need
```suggestion
.Create a constraint requiring `Book` nodes to have unique `title` properties, when there already exists two `Book` nodes with the same `title`
```

Reply:
and similarly for the other examples below, should we specify the label/relationship types as having those properties on unrelated labels/relationship types would not block the constraint creation